src/portable_python/external/xcpython.py

```diff
@@ -21,14 +21,24 @@ class Bdb(ModuleBuilder):
 
     @property
     def url(self):
-        return f"https://ftp.osuosl.org/pub/blfs/conglomeration/db/db-{self.version}.tar.gz"
+        return self.cfg_url(self.version) or f"https://ftp.osuosl.org/pub/blfs/conglomeration/db/db-{self.version}.tar.gz"
```

Comment:
I like this, it's a nice step, we could configure all these urls in the default built-in config (user config naturally overrides it).